ReplayKitExample/ReplayKitExample/ViewController.swift

```diff
@@ -0,0 +1,401 @@
+
+    @IBOutlet weak var spinner: UIActivityIndicatorView!
+    @IBOutlet weak var broadcastButton: UIButton!
+    // Treat this view as generic, since RPSystemBroadcastPickerView is only available on iOS 12.0 and above.
+    @IBOutlet weak var broadcastPickerView: UIView?
+    @IBOutlet weak var conferenceButton: UIButton?
+    @IBOutlet weak var infoLabel: UILabel?
+    @IBOutlet weak var settingsButton: UIBarButtonItem?
+
+    // Conference state.
+    var screenTrack: TVILocalVideoTrack?
+    var videoSource: ReplayKitVideoSource?
+    var conferenceRoom: TVIRoom?
+
+    // Broadcast state. Our extension will capture samples from ReplayKit, and publish them in a Room.
+    var broadcastController: RPBroadcastController?
+
+    var accessToken: String = "TWILIO_ACCESS_TOKEN"
+    let accessTokenUrl = "http://127.0.0.1:5000/"
+
+    static let kStartBroadcastButtonTitle = "Start Broadcast"
+    static let kInProgressBroadcastButtonTitle = "Broadcasting"
+    static let kStopBroadcastButtonTitle = "Stop Broadcast"
+    static let kStartConferenceButtonTitle = "Start Conference"
+    static let kStopConferenceButtonTitle = "Stop Conference"
+    static let kRecordingAvailableInfo = "Ready to share the screen in a Broadcast (extension), or Conference (in-app)."
+    static let kRecordingNotAvailableInfo = "ReplayKit is not available at the moment. Another app might be recording, or AirPlay may be in use."
+
+    // An application has a much higher memory limit than an extension. You may choose to deliver full sized buffers instead.
+    static let kDownscaleBuffers = false
+
+    deinit {
+        NotificationCenter.default.removeObserver(self)
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        broadcastButton.setTitle(ViewController.kStartBroadcastButtonTitle, for: .normal)
+        conferenceButton?.setTitle(ViewController.kStartConferenceButtonTitle, for: .normal)
+        broadcastButton.layer.cornerRadius = 4
+        conferenceButton?.layer.cornerRadius = 4
+
+        // The setter fires an availability changed event, but we check rather than rely on this implementation detail.
+        RPScreenRecorder.shared().delegate = self
+        checkRecordingAvailability()
+
+        NotificationCenter.default.addObserver(forName: NSNotification.Name.UIScreenCapturedDidChange, object: UIScreen.main, queue: OperationQueue.main) { (notification) in
+            if self.broadcastPickerView != nil && self.screenTrack == nil {
+                let isCaptured = UIScreen.main.isCaptured
+                let title = isCaptured ? ViewController.kInProgressBroadcastButtonTitle : ViewController.kStartBroadcastButtonTitle
+                self.broadcastButton.setTitle(title, for: .normal)
+                self.conferenceButton?.isEnabled = !isCaptured
+                isCaptured ? self.spinner.startAnimating() : self.spinner.stopAnimating()
+            }
```

Comment:
I feel we can improve upon this logic ? can `UIScreen.screens.count`help? Again an item for the follow up PR.

Reply:
I was trying to not handle events for other screens that we can't record. I agree there is a lot of work to be done, specifically in ViewController.swift, but it shouldn't block this PR.